event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventlog/EventLogJdbcRepository.java

```diff
@@ -44,35 +51,22 @@
     static final String SQL_FIND_BY_STREAM_ID_AND_SEQUENCE_ID = "SELECT * FROM event_log WHERE stream_id=? AND sequence_id>=? ORDER BY sequence_id ASC";
     static final String SQL_FIND_LATEST_SEQUENCE_ID = "SELECT MAX(sequence_id) FROM event_log WHERE stream_id=?";
     static final String SQL_DISTINCT_STREAM_ID = "SELECT DISTINCT stream_id FROM event_log";
-    static final String SQL_INSERT_EVENT_LOG = "INSERT INTO event_log (id, stream_id, sequence_id, name, metadata, payload/*, date_created*/ ) " +
-            "VALUES(?, ?, ?, ?, ?, ?/*, ?*/)";
 
     private static final String READING_STREAM_ALL_EXCEPTION = "Exception while reading stream";
     private static final String READING_STREAM_EXCEPTION = "Exception while reading stream %s";
     private static final String JNDI_DS_EVENT_STORE_PATTERN = "java:/app/%s/DS.eventstore";
 
-
     /**
-     * Insert the given event into th event log.
+     * Insert the given event into the event log.
      *
      * @param eventLog the event to insert
      * @throws InvalidSequenceIdException if the version already exists or is null.
      */
     public void insert(final EventLog eventLog) throws InvalidSequenceIdException {
-
-        if (eventLog.getSequenceId() == null) {
-            throw new InvalidSequenceIdException(format("Version is null for stream %s", eventLog.getStreamId()));
-        }
-
-        try (final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_INSERT_EVENT_LOG)) {
-            ps.setObject(1, eventLog.getId());
-            ps.setObject(2, eventLog.getStreamId());
-            ps.setLong(3, eventLog.getSequenceId());
-            ps.setString(4, eventLog.getName());
-            ps.setString(5, eventLog.getMetadata());
-            ps.setString(6, eventLog.getPayload());
-            ps.executeUpdate();
+        try (final PreparedStatementWrapper ps = preparedStatementWrapperOf(eventLogInsertionStrategy.insertStatement())) {
+            eventLogInsertionStrategy.insert(ps, eventLog);
         } catch (SQLException e) {
```

Comment:
Yes I know. I put in the comment mainly for the fun of it. However, I usually mark exceptions as final (just in case Matt should do the review) and change any I see. But I agree - it's not important.